### PR TITLE
Automatically sending ride mails

### PR DESCRIPTION
**Pull request by peteruithoven**
Automatically sending ride mails
First email of #20

This is probably not ready yet. 
- [x] I've had to guess the next available id for the mail template. See:
https://github.com/rikva/voko/compare/master...peteruithoven:ride-mails?expand=1#diff-f0bdbbc178ecd7dc0bdd0493855b9233R139
- [x] I'm sure there s a better way to define the `base_url`, should we add `BASE_URL` to `CONSTANCE_CONFIG`? Is there another way? See: https://github.com/rikva/voko/compare/master...peteruithoven:ride-mails?expand=1#diff-5452913ec19e931c6d58d92f02c03e6fR258
- [x] Write the email template

### CODE REVIEW

**Comment by peteruithoven** [comment]
Eerste opzet mail template:
> Beste {{ user.first_name }},
> 
> De bestelronde is zojuist gesloten en je staat ingeroostered om te helpen met transport voor deze ronde.
> Alle informatie over deze rit: {{ base_url }}{% url 'ride' ride.slug %}
> Meer informatie over hoe transport werkt: https://vokoutrecht.nl/transport/
> 
> Bedankt voor het helpen met transport!
> 
> Groene groetjes,
> VOKO Utrecht

**Comment by rikva** [comment]
Just added the template:

image

**Comment by peteruithoven** [comment]
@rikva what are you're thoughts about sharing the base url. Should we add it to `CONSTANCE_CONFIG`?

**Comment by rikva** [comment]
I would suggest adding it to the default context for all mailings, by passing it to `render_mail_template` in `mailing.views.SendMailView#_send_mails`.
At this point the current order round and (recipient) user are passed to every mailing's context.

**Comment by peteruithoven** [comment]
I've used `mailing.helpers`'s `render_mail_template` and `mail_user` because that's how `ordering.models.send_reminder_mails` works. I could add the base_url in `render_mail_template`.
Do you think I should switch to that SendMailView? 

But I mostly meant how do we share this base url throughout the code base? `CONSTANCE_CONFIG` seems like a great option for this?

**Comment by rikva** [comment]
Right, you are already doing what I suggest. Disregard my last comment.

BASE_URL is very static, in my opinion. 
I suggest referencing `settings.BASE_URL` (`from django.conf import settings`)

**Comment by rikva** [comment]
Whereas Constance is meant for more dynamic settings which should be configurable by administrators, IMO.

File: webapp/ordering/models.py [hunk 1]
```diff
@@ -234,6 +240,33 @@ def send_reminder_mails(self):
                 mail_template, user=user, order_round=self)
             mail_user(user, *rendered_template_vars)
 
+    def get_rides(self):
+        from transport.models import Ride
```
Comment (rikva) [inline webapp/ordering/models.py]: Inline import - is this because of a circular dependency?

Comment (rikva) [inline webapp/ordering/models.py]: Are you sure this method is needed? You should be able to reference `self.ride.all()` (the `related_name` on `order_round` FK on `Ride` is called `ride` (ideally would be `rides` :))

Comment (peteruithoven) [reply webapp/ordering/models.py]: Yes indeed. This seemed like a simple solution, especially since I only needed Ride there.

Comment (peteruithoven) [reply webapp/ordering/models.py]: You're absolutely right, that's way easier.

Comment (peteruithoven) [reply webapp/ordering/models.py]: By retrieving the rides from the related field I didn't need this anymore.

**Comment by rikva** [comment]
LGTM. Did you try/test it?

**Comment by peteruithoven** [comment]
Great. Yes, I've just tried it again with the latest changes using `runcrons` and then checking `/tmp/app-messages` and it seems to send the right mails.

**Comment by rikva** [comment]
Deployed!

I've set the `rides_mail_sent` bool of the last order round manually to `True` to prevent a mail being sent.

**Comment by peteruithoven** [comment]
Awesome, and good thinking!

**Pull request merged**